crates/accelerate/src/quantum_circuit/circuit_data.rs

```diff
@@ -213,32 +213,33 @@ impl CircuitData {
         Ok((ty, args, None::<()>, self_.iter()?).into_py(py))
     }
 
-    /// Returns the current sequence of registered :class:`.Qubit`
-    /// instances as a list.
+    /// Returns the current sequence of registered :class:`.Qubit` instances as a list.
     ///
     /// .. note::
     ///
-    ///     This list is not kept in sync with the container.
+    ///     Modifying this list from Python space will invalidate the :class:`CircuitData` data
+    ///     structures.
     ///
     /// Returns:
     ///     list(:class:`.Qubit`): The current sequence of registered qubits.
     #[getter]
-    pub fn qubits(&self, py: Python<'_>) -> PyObject {
-        PyList::new(py, self.qubits.as_ref(py)).into_py(py)
+    pub fn qubits(&self, py: Python<'_>) -> Py<PyList> {
+        self.qubits.clone_ref(py)
     }
 
     /// Returns the current sequence of registered :class:`.Clbit`
     /// instances as a list.
     ///
     /// .. note::
     ///
-    ///     This list is not kept in sync with the container.
+    ///     Modifying this list from Python space will invalidate the :class:`CircuitData` data
```

Comment:
Should we make it a little more direct? Something like:

```
DO NOT modify this list from Python space.
Doing so will invalidate this :class:`CircuitData` instance.
```

Reply:
Sure, I made it a `.. warning::` note and added "Do not" in 203a358d9e.